analyzers/tests/SonarAnalyzer.UnitTest/CFG/Roslyn/BasicBlockTest.cs

```diff
@@ -118,6 +124,88 @@ public void Method(bool condition)
             branch.Predecessors.Should().HaveCount(1);
             assign.Predecessors.Should().HaveCount(1);
             exit.Predecessors.Should().HaveCount(2);
+
+            entry.SuccessorBlocks.Should().HaveCount(1).And.Subject.Single().Should().Be(branch);
+            branch.SuccessorBlocks.Should().HaveCount(2).And.Subject.Should().ContainInOrder(assign, exit);
+            assign.SuccessorBlocks.Should().HaveCount(1).And.Subject.Single().Should().Be(exit);
+            exit.SuccessorBlocks.Should().HaveCount(0);
+        }
+
+        [TestMethod]
+        public void ValidateOperations()
+        {
+            const string code = @"
+public class Sample
+{
+    int Pow(int num, int exponent)
+    {
+        num = num * Pow(num, exponent - 1);
+        return 42;
+    }
+}";
+            var cfg = TestHelper.CompileCfg(code);
+            var entry = cfg.EntryBlock;
+            var exit = cfg.ExitBlock;
+            var body = cfg.Blocks[1];
+            entry.Kind.Should().Be(BasicBlockKind.Entry);
+            entry.Operations.Should().BeEmpty();
+            exit.Kind.Should().Be(BasicBlockKind.Exit);
+            exit.Operations.Should().BeEmpty();
+            body.Kind.Should().Be(BasicBlockKind.Block);
+            body.Operations.Should().HaveCount(1).And.Subject.Single().Should().BeAssignableTo<IExpressionStatementOperation>();
+            body.BranchValue.Should().BeAssignableTo<ILiteralOperation>();
+            body.OperationsAndBranchValue.Should().OnlyContainInOrder(body.Operations[0], body.BranchValue);
+        }
+
+        [TestMethod]
+        public void ValidateSwitchExpressionCase()
+        {
+            const string code = @"
+public class Sample
+{
+    public int Method(bool condition) =>
+        condition switch
+        {
+            true => 42,
+            _ => 43
+        };
+}";
+            var cfg = TestHelper.CompileCfg(code);
+            /*
+             *
+             *         Block 1
+             *        true => 42
+             * Else  /          \    WhenFalse
+             *      /            \
+             *     /              \
+             *  Block 2          Block 3
+             *   => 42           _ => 43
+             *    |                |            WhenFalse from Block 3 - Block 5 should not be reachable
+             *    |                |
+             *    |              Block 4        Block 5
```

Comment:
The branch is missing, so it's hard to understand
```suggestion
             *    |                |       \      WhenFalse from Block 3 - Block 5 should not be reachable
             *    |                |        \
             *    |              Block 4        Block 5
```